src/integration_tests/camera_take_photo.cpp

```diff
@@ -1,43 +1,57 @@
-#include <iostream>
-#include <functional>
-#include <vector>
-#include <atomic>
-#include <future>
 #include "integration_test_helper.h"
 #include "mavsdk.h"
 #include "system.h"
-#include "camera_test_helpers.h"
+#include "plugins/camera/camera.h"
+#include "plugins/camera_server/camera_server.h"
 
 using namespace mavsdk;
 
 static void receive_capture_info(Camera::CaptureInfo capture_info, bool& received_capture_info);
 
 TEST(CameraTest, TakePhotoSingle)
 {
-    Mavsdk mavsdk;
+    Mavsdk mavsdk_groundstation;
+    mavsdk_groundstation.set_configuration(
+        Mavsdk::Configuration{Mavsdk::Configuration::UsageType::GroundStation});
 
-    ConnectionResult ret = mavsdk.add_udp_connection();
-    ASSERT_EQ(ret, ConnectionResult::Success);
+    Mavsdk mavsdk_camera;
+    mavsdk_camera.set_configuration(
+        Mavsdk::Configuration{Mavsdk::Configuration::UsageType::Camera});
+
+    ASSERT_EQ(mavsdk_groundstation.add_any_connection("udp://:17000"), ConnectionResult::Success);
+    ASSERT_EQ(mavsdk_camera.add_any_connection("udp://127.0.0.1:17000"), ConnectionResult::Success);
+
+    auto camera_server =
+        CameraServer{mavsdk_camera.server_component_by_type(Mavsdk::ServerComponentType::Camera)};
+    camera_server.subscribe_take_photo([&camera_server](int32_t index) {
+        LogInfo() << "Let's take photo " << index;
+
+        CameraServer::CaptureInfo info;
+        info.index = index;
+        info.is_success = true;
+
+        camera_server.respond_take_photo(CameraServer::TakePhotoFeedback::Ok, info);
+    });
 
     // Wait for system to connect via heartbeat.
     std::this_thread::sleep_for(std::chrono::seconds(2));
 
-    auto system = mavsdk.systems().at(0);
+    ASSERT_EQ(mavsdk_groundstation.systems().size(), 1);
+    auto system = mavsdk_groundstation.systems().at(0);
     ASSERT_TRUE(system->has_camera());
-    auto camera = std::make_shared<Camera>(system);
 
-    std::this_thread::sleep_for(std::chrono::seconds(1));
+    auto camera = Camera{system};
 
     // We want to take the picture in photo mode.
-    EXPECT_EQ(camera->set_mode(Camera::Mode::Photo), Camera::Result::Success);
+    // EXPECT_EQ(camera.set_mode(Camera::Mode::Photo), Camera::Result::Success);
```

Comment:
Shouldn't it be enabled?

```suggestion
    EXPECT_EQ(camera.set_mode(Camera::Mode::Photo), Camera::Result::Success);
```

Reply:
Probably yes, not sure what happened there.